sklearn/utils/validation.py

```diff
@@ -581,6 +587,15 @@ def check_array(array, accept_sparse=False, accept_large_sparse=True,
     if copy and np.may_share_memory(array, array_orig):
         array = np.array(array, dtype=dtype, order=order)
 
+    if warn_on_dtype and dtypes_orig is not None and {array.dtype} != \
+            set(dtypes_orig):
```

Comment:
@jnothman does sklearn have some style rules on avoiding `\` in new code for line continuation?